* this fix a problem when setting the essid, the last character is removed
* it surely will have to be updated to match kernel wireless extensions version
* I (FoX) contacted the author about this "issue", still waiting for an answer

diff -Naur wireless_tools.28.orig/iwconfig.c wireless_tools.28/iwconfig.c
--- wireless_tools.28.orig/iwconfig.c	2007-01-04 19:17:19.000000000 +0100
+++ wireless_tools.28/iwconfig.c	2007-01-04 19:17:35.000000000 +0100
@@ -926,7 +926,7 @@
 	  /* Finally set the ESSID value */
 	  wrq.u.essid.pointer = (caddr_t) essid;
 	  wrq.u.essid.length = strlen(essid) + 1;
-	  if(we_kernel_version > 20)
+	  if(we_kernel_version > 21)
 	    wrq.u.essid.length--;
 	  IW_SET_EXT_ERR(skfd, ifname, SIOCSIWESSID, &wrq,
 			 "Set ESSID");
